wpilibc/src/main/native/include/frc/SendableSpeedController.h

```diff
@@ -0,0 +1,24 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include "frc/SpeedController.h"
+#include "frc/smartdashboard/SendableBase.h"
+
+namespace frc {
+
+/**
+ * Interface for speed controlling devices that can be displayed on a dashboard.
+ */
+class SendableSpeedController : public virtual SendableBase,
```

Comment:
Shouldn't this derive from Sendable, not SendableBase?  Could that get rid of the need for virtual inheritance?

Reply:
Or on second thought, maybe it's better for it to derive from SendableBase?  Again, trying to avoid virtual inheritance.

Reply:
`PWM` also derives from `SendableBase`, which makes this difficult. As it is, the current hierarchy (that I'm trying to minimize changes to, in hopes that this gets accepted) already has a diamond on `PWMSpeedController`, and the virtual inheritance is the only way to *properly* manage that without ending up with two copies of `Sendable` in the hierarchy.

Deriving from `Sendable` wouldn't make any difference in this scenario. I honestly don't know how it works properly right now, since PWMSpeedController has two `Sendable` in its class hierarchy already. Using virtual inheritance at least combines the two into one instance.

Reply:
`PWM` derives from `SendableBase` but not `SpeedController`. `PWMSpeedController` derives from `PWM` and `SpeedController`. Without poking at the class hierarchy more substantially (which I'm hesitant to do) I don't quite see a way to make `PWMSpeedController` (and thus all derived classes) implement the new interface, without which this is very limited functionality.

Using virtual inheritance isn't ideal, but it at least limits the `SendableBase` instance to just one copy in the class hierarchy.

Reply:
We could remove the diamond inheritance problem with `PWMSpeedController` by not having `PWM` inherit from SendableBase. To avoid breaking anyone relying on this behavior we could rename `PWM` to `_PWM` (or some other name) and make `PWM` inherit from `_PWM` and `SendableBase`. `PWMSpeedController would then inherit from `_PWM`, avoiding the diamond inheritance. It's a bit ugly, but then that change in behavior could be deprecated in the future.

Reply:
PWMBase?

Reply:
Removed